chainer/functions/loss/negative_sampling.py

```diff
@@ -102,14 +118,22 @@ def backward_cpu(self, inputs, grads):
 
         gx = numpy.zeros_like(x)
         gW = numpy.zeros_like(W)
-        for i, (ix, k) in enumerate(six.moves.zip(
-                x[self.ignore_mask], self.samples[self.ignore_mask])):
+        for i, active in enumerate(self.ignore_mask):
```

Comment:
I think `for i in numpy.arange(len(self.ignore_mask))[~self.ignore_mask]` is faster because it has smaller loop counts.